### DISCUSSION TITLE
trouble with conditionals

### DISCUSSION THREAD

**Discussion by mundorfd** [ttscoff/bunch-docs#147] [Q&A] [answered]
I am trying to open particular folders in the Finder, URLs and URL schemes based on a variable. It isn't working... I verified through logging that the variable was getting set, but it doesn't seem to be executing the item in the conditional based on that variable. Here's the code (with some name changes...)

```
---
title: Debug Course Development
menu order: 1
---

course = ?[ernie, bert, oscar] "Which environment are you working in?"

%Finder
- XX
if course is "ernie"
	- /Volumes/GoogleDrive/My Drive/ernie
else if course is "bert"
	- /Volumes/GoogleDrive/Shared drives/bert
else if course is "oscar"
	- ~/Box/Courses Shared Folder/oscar

if course is "ernie"
	https://www.ernie.com
else if course is "bert"
	https://www.bert.com	
else if course is "oscar"
	https://www.oscar.com

if course is "ernie"
	omnifocus:///task/ernie
else if course is "bert"
	omnifocus:///task/bert
	obsidian://open?vault=vault_deb&file=bert
else if course is "oscar"
	omnifocus:///task/oscar
	obsidian://open?vault=vault_deb&file=oscar

# tell application "Moom" to arrange windows according to snapshot "sesamestreet"
	<moom_sesamestreet.snippet
```

I'm very new to this tool... hoping someone can help!

**Comment by ttscoff** [comment, marked as answer]
first, it looks like you're missing the `end if` for those blocks. In that first block, you need to re-include Finder in each condition. It might actually work like you have it, with the file lines inside conditionals, but it wasn't designed that way and I can't be sure of that. Never actually tried it. What I'd recommend, though:

```
if course is "ernie"
	Finder
	- /Volumes/GoogleDrive/My Drive/ernie
else if course is "bert"
	Finder
	- /Volumes/GoogleDrive/Shared drives/bert
else if course is "oscar"
	Finder
	- ~/Box/Courses Shared Folder/oscar
end if
```

I think with the appropriate `end if` on the other two blocks they should work as is. You could combine them all, though...

```
if course is "ernie"
	Finder
	- /Volumes/GoogleDrive/My Drive/ernie
	https://www.ernie.com
	omnifocus:///task/ernie
else if course is "bert"
	# etc. etc.
end if
```

Let me know if that helps.

**Reply by mundorfd** [reply]
Yes, and now I feel silly. Thank you so much for your quick help. Sending more coffee $.
